### DISCUSSION TITLE
Improve the UX for download users: Batch download of models, sample images, and README

### DISCUSSION THREAD

**Discussion by WarriorMama777** [civitai/civitai#222] [Ideas]
Clicking the download button only gives a learning model, which ruins the graphical and detailed prompt information that the uploader has built up. They can only be saved by the user using 'right-click and save'.
The download folder is quickly cluttered with huge files with no thumbnails.
It should be possible to save the training model in one go, including the information on the page.

My ideal download flow for users would be as follows.
(1) The main download button (top right) downloads all the data contained on the page in one batch (.ckpt or LoRA etc... ,README.md, SampleImages, etc... )
(2) The download button next to version downloads 1. the model data corresponding to the version, 2. README.md, 3. SampleImages (etc...) Download them all at once.

**Comment by tehrobber** [comment]
There's a user script that adds a button which downloads all the preview images as a ZIP - https://github.com/tehrobber/civitai-one-click-dl/

**Reply by shaneholloman** [reply]
https://github.com/tehrobber/civitai-one-click-dl/issues awesome idea!

**Reply by WASasquatch** [reply]
Did you get it to work? It seems like a crazy userscript using the civitai API? Probably best to just use javascript and parse the page without any hooks. 

If I did this I'd probably just loop the version containers, scrape visible images, as max on screen before view more is plenty for model previews.

**Reply by shaneholloman** [reply]
@WASasquatch Jordan,  yes Rob got it working. He's tweaking on it a bit still

**Comment by ghost** [comment]
I made a browser extension, it will download the model, the first 8 images, and the image generation data and description to a text file. It only works for chrome based browsers and has to be used in development mode atm. https://github.com/poorair/Civitai-Bulk-Download

**Comment by JohnMadden123456** [comment]
Not exactly what you looking for, but I'm using combination of  [Webrecorder ArchiveWeb.page](https://chrome.google.com/webstore/detail/webrecorder-archivewebpag/fpeoodllldobpkbkabpblcfaogecpndd) which creates `wacz` archive file and [ReplayWeb.page](https://github.com/webrecorder/replayweb.page) to view those archives. It has some support for automation, but I haven't checked that one.
